File: src/geometryService/cpp/stractNet/IntegerFrag.cpp
///////////////////////////////////////////////////////////
//  IntegerFrag.cpp
//  Implementation of the Class IntegerFrag
//  Created on:      06-Aug-2008 7:51:52 AM
//  Original author: david.h.loman
///////////////////////////////////////////////////////////
#include "IntegerFrag.h"
///////////////////////////////////////////////////////////////////////
/*
 *
 * xSTRUCTORS
 *
 */
///////////////////////////////////////////////////////////////////////
IntegerFrag::~IntegerFrag(){}
IntegerFrag::IntegerFrag(int in){
	MsgFrag("IntegerFrag", in, MsgFrag::mftInteger);
}
IntegerFrag::IntegerFrag(DataInputStream& in){
	MsgFrag("IntegerFrag", in, MsgFrag::mftInteger);
}
///////////////////////////////////////////////////////////////////////
/*
 *
 * METHODS
 *
 */
///////////////////////////////////////////////////////////////////////
void IntegerFrag::Deserialize(DataInputStream& in){
	try {
		// Read Integer
		this->field = in.readInt();
	} catch (IOException ioe) {
		SNRoot::err("IOException in Deserialize()", 0);
	}
	return;
}
void IntegerFrag::Serialize_Specific(DataOutputStream& out){
	try {
		// write in new data
		out.writeInt(this->field);
	} catch (IOException ioe) {
		SNRoot::err("IOException in Serialize()", 0);
	}
	return;
}

File: src/superceded_GS/java/stractNet/src/stractNet/messaging/msgs/frags/IntegerFrag.java
package stractNet.messaging.msgs.frags;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

public class IntegerFrag extends MsgFrag<Integer> {
	public IntegerFrag(int in) {
		super("IntegerFrag", in, MsgFrag.mftInteger);
	}

	public IntegerFrag(DataInputStream in) {
		super("IntegerFrag", in, MsgFrag.mftInteger);
	}

	@Override
	public void Deserialize(DataInputStream in) {
		try {
			// Read Integer
			this.field = in.readInt();
		} catch (IOException ioe) {
			this.err("IOException in Deserialize()", 0);
		}
		return;
	}

	@Override
	protected void Serialize_Specific(DataOutputStream out) {
		try {
			// write in new data
			out.writeInt(this.field);
		} catch (IOException ioe) {
			this.err("IOException in Serialize()", 0);
		}

		return;
	}

}
